pkg/handlers/generic/lifecycle/clusterautoscaler/strategy_helmaddon.go

```diff
@@ -0,0 +1,136 @@
+// Copyright 2023 D2iQ, Inc. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package clusterautoscaler
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/go-logr/logr"
+	"github.com/spf13/pflag"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	capiv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	runtimehooksv1 "sigs.k8s.io/cluster-api/exp/runtime/hooks/api/v1alpha1"
+	ctrlclient "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	caaphv1 "github.com/d2iq-labs/capi-runtime-extensions/api/external/sigs.k8s.io/cluster-api-addon-provider-helm/api/v1alpha1"
+	"github.com/d2iq-labs/capi-runtime-extensions/common/pkg/k8s/client"
+)
+
+const (
+	defaultHelmRepositoryURL       = "https://kubernetes.github.io/autoscaler"
+	defaultHelmChartVersion        = "9.35.0"
+	defaultHelmChartName           = "cluster-autoscaler"
+	defaultHelmReleaseNameTemplate = "cluster-autoscaler-%s"
+)
+
+type helmAddonConfig struct {
+	defaultValuesTemplateConfigMapName string
+}
+
+func (c *helmAddonConfig) AddFlags(prefix string, flags *pflag.FlagSet) {
+	flags.StringVar(
+		&c.defaultValuesTemplateConfigMapName,
+		prefix+".default-values-template-configmap-name",
+		"default-cluster-autoscaler-helm-values-template",
+		"default values ConfigMap name",
+	)
+}
+
+type helmAddonStrategy struct {
+	config helmAddonConfig
+
+	client ctrlclient.Client
+}
+
+func (s helmAddonStrategy) apply(
+	ctx context.Context,
+	req *runtimehooksv1.AfterControlPlaneInitializedRequest,
+	defaultsNamespace string,
+	log logr.Logger,
+) error {
+	log.Info("Retrieving cluster-autoscaler installation values template for cluster")
+	valuesTemplateConfigMap, err := s.retrieveValuesTemplateConfigMap(ctx, defaultsNamespace)
+	if err != nil {
+		return fmt.Errorf(
+			"failed to retrieve cluster-autoscaler installation values template ConfigMap for cluster: %w",
+			err,
+		)
+	}
+
+	cluster := &req.Cluster
+
+	data := templateData(valuesTemplateConfigMap.Data, cluster.Name, cluster.Namespace)
+	values := data["values.yaml"]
```

Comment:
No need for this if we can use the CAAPH templating.
```suggestion
	values := valuesTemplateConfigMap["values.yaml"]
```

Reply:
Done.